vision_agent_tools/tools/text_to_object_detection.py

```diff
@@ -60,12 +116,33 @@ def __call__(
         if image is not None and video is not None:
             raise ValueError("Only one of 'image' or 'video' can be provided.")
 
-        if image is not None:
-            prediction = self.model(image=image, prompts=prompts)
-        if video is not None:
-            prediction = self.model(video=video, prompts=prompts)
-
-        output = TextToObjectDetectionOutput(output=prediction)
+        if self.model == TextToObjectDetectionModel.OWLV2:
+            if image is not None:
+                prediction_output = self.model(image=image, prompts=prompts)
+            if video is not None:
+                prediction_output = self.model(video=video, prompts=prompts)
+        elif self.model == TextToObjectDetectionModel.FLORENCEV2:
+            od_task = PromptTask.OBJECT_DETECTION
+            if image is not None:
+                prediction_output = self.model(
+                    image=image,
+                    task=od_task,
+                )
+            if video is not None:
+                prediction_output = self.model(
+                    video=video,
+                    task=od_task,
+                )
+
+            # Prediction should be a list of lists of ODResponseData objects
+            # We need to convert the output to the format that is used in the playground-tools
+            std_output = []
+            for pred in prediction_output:
```

Comment:
There is a test error on `'prediction_output' referenced before assignment` here

Reply:
working on it